components/frontend-react/src/components/UserSignup.js

```diff
@@ -42,19 +42,53 @@ const useStyles = makeStyles(theme => ({
 
 const UserSignup = (props) => {
 
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setconfirmPassword] = useState('');
+
+  const handleTextFields = (event) => {
```

Comment:
@agentmilindu Shall we use `formik` to handle form data and validations?

Reply:
yeah, that sounds good